test/hooks/useSelector.spec.js

```diff
@@ -350,13 +350,11 @@ describe('React', () => {
             }
           )
 
-          let sawInconsistentState = false
+          let lastRenderInconsistentState = false
 
           const Child = ({ parentCount }) => {
             const result = useSelector(({ count }) => {
-              if (count !== parentCount) {
-                sawInconsistentState = true
-              }
+              lastRenderInconsistentState = count !== parentCount
```

Comment:
I don't understand this change. The whole purpose of this test was to ensure the selector never sees inconsistent state, not just on the latest render.

Reply:
Because we NO more read the last state in the render phase, the "parent-child-issue" is resolved with 2 render, as every change of the store make 2 event-callback one on the parent and one on the child.
Before the issue was resolved with the "hack" of reading always the fresh Store state, even if we are on the "preview" scheduled refresh came out from "checkForUpdate" callback.

That hack while resolve the "parent-issue" with 1 render less, broke the changes normal queue on the the componente, that produce tearing on CM where every component could rendered in a random way, jumping the "changhes queue"

Reply:
please try the demo of @dai-shi with my version and with old one. you will see the differences on a fast store changes, with slow render.